### PR TITLE
Fix fancytree multiselect level issue

### PR DESCRIPTION
**Pull request by deanL-zuiderlicht**
Nodes more than two levels deep do not get preselected on load even though they are present on the ajaxData object.
By checking the childNodes on expanding a node we can select the nodes afterwards. Added the collapse callback because unchecking nodes kept preselecting the nodes again due to the ajaxData.selected array. By removing unchecked key from the array this issue is also resolved.

### CODE REVIEW

File: assets/cms/js/tree.js [hunk 1]
```diff
@@ -218,7 +218,25 @@ ConcreteTree.prototype = {
...
                 return true
             },
-
+            expand: function(event, data) {
+                //only if 'selected' array is available execute code.
+                if (options.ajaxData.selected) {
+                    //since the expand options is triggered we can asume that there are children.
+                    data.node.children.forEach(function(nodeChild){
+                        if (options.ajaxData.selected.includes(nodeChild.key)) {
+                            nodeChild.setSelected(!0);
```
Comment (mlocati) [inline assets/cms/js/tree.js:227]: What about using `true` instead of `!0` ?

Comment (deanL-zuiderlicht) [reply assets/cms/js/tree.js:227]: That is better indeed. I tested with the minified one from the core so I copied it from there. I will change this to `true`.

**Comment by aembler** [comment]
This will be great when it can be merged in. Can you update the code per mlocati's feedback? This issue has been annoying for so long!

**Comment by deanL-zuiderlicht** [comment]
Last week I committed a change. Are there any other changes necessary?

**Comment by aembler** [comment]
This isn't doing what I thought it was doing. I assumed this was to fix a long-standing issue where the sitemap, when used in certain forms like page types, etc... would not open to show all the nodes to the selected node when the node was deep in the tree. It doesn't appear like this fixes that issue. Could you explain what this code fixes a bit more clearly? How to reproduce the issue, etc.. ?

**Comment by deanL-zuiderlicht** [comment]
This should fix the issue of deeper laying nodes in FancyTree indeed. So more than two levels deep. To reproduce you can add a page attribute that allows multi select of pages. When selecting deeper laying nodes these should appear unchecked after saving.
Another option is to add a block with the fancytree multiselect options like this
`<?php echo Core::make("helper/form/page_selector")->selectMultipleFromSitemap($view->field('pages'), $pages); ?>`

To be honest i tested my changes in an existing site of ours. Within the minified cms.js file. So I will check out my forked branche and test it properly.

**Comment by deanL-zuiderlicht** [comment]
How can i compile the code locally? When i run the laravel mix script i get a notice that the app.js and app.scss are missing. Which is correct obviously. Running storybook also gives an error though. Complaining about deprecation and dependencies.

**Comment by deanL-zuiderlicht** [comment]
I added an other commit to parse the int from the nodeKey because the selected array contains integers but the node has a string for an ID. Could you review?

**Comment by deanL-zuiderlicht** [comment]
Hi, sorry for the spam but we have clients depending on this functionality so I would greatly appreciate it if you could take a look at my latest changes.
Thanks.

**Comment by deanL-zuiderlicht** [comment]
Hi,
I saw that i was mentioned in the release of 9.2.1. Thanks for that! Appreciate it.
Just installed the latest version and i was testing the functionality but unfortunately it does not seem to be working..
Did it work for you? 
I placed a debugger in the 'expand' function within cms.js but it does not get triggered.

**Comment by aembler** [comment]
I was never able to reproduce the issue personally.

**Pull request merged**